code/modules/food_and_drinks/restaurant/_venue.dm

```diff
@@ -39,7 +38,27 @@
 
 ///Spawns a new customer at the portal
 /datum/venue/proc/create_new_customer()
-	var/mob/living/simple_animal/robot_customer/new_customer = new /mob/living/simple_animal/robot_customer(get_turf(restaurant_portal), pickweight(customer_types), src)
+	var/list/customer_types_to_choose = customer_types
+	var/datum/customer_data/customer_type
+
+	// In practice, the list will never run out, but this is for sanity.
+	while (customer_types_to_choose.len)
+		customer_type = pickweight(customer_types_to_choose)
+
+		var/datum/customer_data/customer = SSrestaurant.all_customers[customer_type]
+		if (customer.can_use(src))
+			break
+
+		// Only copy the list once, so that we're not mutating ourselves.
+		if (customer_types_to_choose == customer_types)
+			customer_types_to_choose = customer_types.Copy()
+
```

Comment:
Why not just do the copy initially instead of having this check?

Reply:
Because most of the time, the first customer we get is going to be usable. Performing a copy every time, when in (literally) 99% of cases it will be unnecessary is wasteful.

Reply:
How hot is this code? is the cleanliness worth the performance?

Reply:
It's not hot, but I don't think this is much harder to read.

Reply:
aight